core/io/net_socket.h

```diff
@@ -52,21 +52,68 @@ class NetSocket : public RefCounted {
 		TYPE_UDP,
 	};
 
-	virtual Error open(Type p_type, IP::Type &ip_type) = 0;
+	enum class Family {
+		NONE,
+		INET,
+		UNIX,
+	};
+
+	class Address {
```

Comment:
This allows the `NetSocket` to distinguish the address family as either INET or UNIX